lib/rack/session/redis.rb

```diff
@@ -78,6 +81,12 @@ def with_lock(req, default=nil)
       def with(&block)
         @conn.with(&block)
       end
+
+      private
+
+      def get_session_with_fallback(sid)
+        with { |c| c.get(sid.private_id) || c.get(sid.public_id) }
```

Comment:
This had the effect of invalidating all existing logins because `sid.public_id` was essentially used in Rack v2.0.x. Was this intentional?

Reply:
Yes. Sessions identified by public ID only would still be susceptible to the attack, so it's best to make users log in again in order to ensure everyone is on the same page.